app/models/reservation.rb

```diff
@@ -189,13 +189,32 @@ def can_cancel?
   end
 
   def can_customer_edit?
-    !canceled? && !complete? && before_lock_window?
+    !canceled? && !complete? && (reserve_start_at_editable? || reserve_end_at_editable?)
+  end
+
+  def reserve_start_at_editable?
+    before_lock_window? && actual_start_at.blank?
+  end
+
+  def reserve_end_at_editable?
+    outside_lock_window? && Time.zone.now <= reserve_end_at && next_duration_available? && actual_end_at.blank?
+  end
+
+  def next_duration_available?
```

Comment:
Yea, I'll switch to use that instead and add a spec for the end of day situation.